src/lib/api/axiosInstance.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @file axiosInstance.js
+ * Exports a method that assists in communicating with EditVR's API.
+ */
+
+import axios from 'axios';
+
+import { apiURL } from '../../config';
+import {
+  ERROR_API_CONNECTION_FAILED,
+  ERROR_API_FORBIDDEN,
+  ERROR_API_GENERAL,
+  ERROR_API_NOT_FOUND,
+  ERROR_API_LOGIN_FAILED,
+  ERROR_API_BAD_REQUEST,
+  AXIOS_ERROR_NETWORK,
+  AXIOS_ERROR_400,
+  AXIOS_ERROR_401,
+  AXIOS_ERROR_403,
+  AXIOS_ERROR_404
+} from '../../constants';
+
+/**
+ * Creates an Axios instance configured to connect to the EditVR API.
+ *
+ * @param {object} authentication - Auth config for this method.
+ * @param {string} authentication.accessToken - Access token for this request.
+ * @param {string} authentication.csrfToken - CSRF-Token token for this request.
+ * @param {string} format - API format. Default is jsonapi.
+ *
+ * @returns {object} - Axios instance.
+ */
+const axiosInstance = (
+  { accessToken = null, csrfToken = null } = {},
+  format = 'jsonapi'
+) => {
+  const config = {
+    baseURL: `${apiURL}/${format || ''}`,
+    timeout: 30 * 1000,
+    headers: {},
+    withCredentials: false,
+    crossdomain: true
+  };
+
+  // If an access token is specified, add it to the header.
+  if (accessToken) {
+    config.headers.Authorization = `Bearer ${accessToken}`;
+  }
+
+  // If a csrfToken is specified, add it to the header.
+  if (csrfToken) {
+    config.headers['X-CSRF-Token'] = csrfToken;
+  }
+
+  // If the format is set to jsonapi, add content-type headers.
+  if (format === 'jsonapi') {
+    config.headers['Content-Type'] = 'application/vnd.api+json';
+  }
+
+  // Generate an axios instance.
+  const instance = axios.create(config);
+
+  // Add an interceptor that cleans up and denormalizes API responses, and
+  // throws coherent errors.
+  instance.interceptors.response.use(
+    response => {
+      let r = response;
+
+      // TODO: This method will soon contain logic for denormalizing JSON-API
+      // responses, when the content type is correct.
```

Comment:
That'll be the next PR :)

Reply:
You may want to keep the denormalizing out of axios and leave it in the action sagas because what if you need to get a non-jsonapi response someday? easy to fix later tho.

Reply:
I added this in #3: https://github.com/EditVR/editvr-frontend/pull/3/files#diff-4f2cf9396637de3e1e740654e0138084R74